src/main/java/org/candlepin/subscriptions/tally/AccountUsageCalculationCache.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.tally;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.json.Event;
+import org.candlepin.subscriptions.util.DateRange;
+
+/**
+ * An object that temporarily caches instances of {@link AccountUsageCalculation} objects and Maps
+ * them by {@link OffsetDateTime}. This object is used to cache any current usage calculations for
+ * an associated usage timestamp (Event.timestamp) while the hourly tally is running. This ensures
+ * that the {@link MetricUsageCollector} will only need to initialize the usage calculation from the
+ * DB if it is not already in the cache, avoiding the need to loading the same calculation for each
+ * batch of events.
+ */
+@Slf4j
+public class AccountUsageCalculationCache {
```

Comment:
What are the bounds on this cache? Would it make sense to have it be implemented w/ jcache in future work? or become tuneable?

Reply:
In most cases, there should only be less than a handful of account calculations cached. Outside of cost, only the hour being processed. Personally, I think that using JCache would be overkill for what this is doing.